docs/src/main/docs/security/05_extensibility.adoc

```diff
@@ -0,0 +1,129 @@
+Implementation can update outgoing message headers to handle
+security for an outgoing request (e.g. identity propagation, mapping etc.).
+
+=== Audit provider
+To create a custom audit provider, create a class that implements
+`io.helidon.security.spi.AuditProvider`.
+Security component feeds each audit provider all messages from all components
+that invoke audit method on "Security" class, including internal audit events
+pre-configured in the component itself (e.g. authentication, authorization
+ events).
+
+Implementation may do whatever desired with these messages, e.g.:
+
+* filter them
+* log them
+* store them to a database
+* forward them to an audit component
+* discard them
+
+== Provider selection policy
+Each request is processed by a single authentication and/or authorization
+ provider. The selection policy provides the security component information about
+ which provider to use. Out of the box, there are three policies:
+
+1. "First" policy - first configured provider (or explicitly defined default
+ provider) is used by default, if a named provider is requested, it would be used
+2. "Composite" policy - this policy allows for a sequence of providers to be
+ executed (e.g. one request may have more than one provider) - used for example to
+ resolve service and user authentication
+3. "Class" policy - this allows usage of a custom policy defined by fully
+ qualified class name
+
+To create a custom provider selection policy, create a class that implements
+"io.helidon.security.spi.ProviderSelectionPolicy".
+
+== Framework integration
+Security provides integration with Helidon WebServer (`helidon-security-integration-webserver`)
+ and with Jersey (`helidon-security-integration-jersey`).
+
+These integrations use APIs that are available to integrate any framework/application (for example
+we could integrate security with messaging, such as JMS).
+
+To create a new integration, an instance of `Security` class is needed, as it handles
+all configured providers. Usually a single `Security` instance is used for an application.
+
+`Security` is then used to create an instance of `SecurityContext`, that should be used
+for interaction with a single user - in HTTP, we create a single `SecurityContext` per
+request.
+
+`SecurityContext` is used to invoke authentication, authorization, and outbound security requests.
+
+Helidon security also defines a set of annotations:
+1. @Authenticated - access to resources must follow authentication rules defined by the annotation
+2. @Authorized - access to resources must follow authorization rules defined by the annotation
+3. @Audited - to configure auditing
+4. Annotations from javax.annotation-api, such as (optional):
+    1. RolesAllowed
+    2. DenyAll
+    3. PermitAll
+
+If the protected resources (in Helidon MP, these are JAX-RS resource classes and methods) can
+be annotated, the integration should at least support the annotations mentioned above.
```

Comment:
By "should at least support" did you mean that our security component "will" support these annotations at a minimum?  Or did you mean that users should use the security component in a way that at least these annotations are supported? This part is not clear to me from the current text.